src/QsCompiler/CompilationManager/EditorSupport/SymbolInfo.cs

```diff
@@ -122,6 +127,66 @@ bool OverlapsWithPosition(Range symRange)
                 usedLiterals: overlappingLiterals.ToImmutableHashSet());
         }
 
+        /// <summary>
+        /// Returns the local declarations that are in scope at <paramref name="position"/>. Assumes that
+        /// <paramref name="scope"/> contains <paramref name="position"/>.
+        /// </summary>
+        /// <param name="scope">The scope to look in.</param>
+        /// <param name="position">The position in the scope to look at.</param>
+        /// <param name="inclusive">
+        /// True if locals declared by the statement or expression that contains <paramref name="position"/> should be
+        /// included.
+        /// </param>
+        /// <returns>The local declarations.</returns>
+        internal static LocalDeclarations LocalsInScope(QsScope scope, Position position, bool inclusive)
+        {
+            var (parent, statementsBefore, _) = SplitStatementsByPosition(scope, position);
+            var declarationsBefore = parent.KnownSymbols.Variables
+                .Concat(statementsBefore.SkipLast(1).SelectMany(s => s.SymbolDeclarations.Variables));
+            var currentStatement = statementsBefore.LastOrDefault();
+
+            if (!inclusive || currentStatement is null)
+            {
+                return new LocalDeclarations(declarationsBefore.ToImmutableArray());
+            }
+
+            var expressionDeclarations = currentStatement.Statement switch
+            {
+                QsStatementKind.QsExpressionStatement expression => ExpressionDeclarations(expression.Item),
+                QsStatementKind.QsReturnStatement @return => ExpressionDeclarations(@return.Item),
+                QsStatementKind.QsFailStatement fail => ExpressionDeclarations(fail.Item),
+                QsStatementKind.QsVariableDeclaration declaration => ExpressionDeclarations(declaration.Item.Rhs),
+                QsStatementKind.QsValueUpdate update => ExpressionDeclarations(update.Item.Rhs),
+                QsStatementKind.QsConditionalStatement cond => CondExpressionDeclarations(cond),
+                QsStatementKind.QsForStatement @for => ExpressionDeclarations(@for.Item.IterationValues),
+                QsStatementKind.QsWhileStatement @while => ExpressionDeclarations(@while.Item.Condition),
+                QsStatementKind.QsRepeatStatement repeat => ExpressionDeclarations(repeat.Item.SuccessCondition),
+                _ => Enumerable.Empty<LocalVariableDeclaration<string>>(),
```

Comment:
While unlikely, a qubit scope could have a variable declaration.

Reply:
I don't think so; a qubit initializer is not an expression and cannot contain a lambda.

Reply:
The qubit array initializer takes an expression, and that expression could contain a lambda. Think of some function that takes a function as argument an returns the number of qubits. While not very common, 
a) the compiler needs to cover all cases not just the common ones
b) it is not unreasonable to have something like the following e.g. in a Chemistry application:
```
let qs = Qubit[RequiredNumberOfQubits(x => x *x, ...)];
```

Reply:
I don't mean that the compiler shouldn't handle uncommon cases; I just forgot about the qubit array length expression. Thanks for explaining.